presto-main/src/main/resources/webapp/src/utils.js

```diff
@@ -96,34 +96,39 @@ export function getStageStateColor(stage: any): string
     }
 }
 
-// This relies on the fact that BasicQueryInfo and QueryInfo have all the fields
-// necessary to compute this string, and that these fields are consistently named.
-export function getHumanReadableState(query: any): string
+export function getHumanReadableState(
+    queryState: string,
+    scheduled: boolean,
+    fullyBlocked: boolean,
+    blockedReasons: Array<mixed>,
+    memoryPool: string,
+    errorType: string,
+    errorCodeName: string): string
 {
-    if (query.state === "RUNNING") {
+    if (queryState === "RUNNING") {
         let title = "RUNNING";
 
-        if (query.scheduled && query.queryStats.totalDrivers > 0 && query.queryStats.runningDrivers >= 0) {
-            if (query.queryStats.fullyBlocked) {
+        if (scheduled) {
+            if (fullyBlocked) {
                 title = "BLOCKED";
 
-                if (query.queryStats.blockedReasons && query.queryStats.blockedReasons.length > 0) {
-                    title += " (" + query.queryStats.blockedReasons.join(", ") + ")";
+                if (blockedReasons && blockedReasons.length > 0) {
```

Comment:
Does it make sense to truncate this info to make sure it doesn't end up making UI non usable if it's too long?

Reply:
And we can have all the reasons in the query view page where user can view single query.

Reply:
to contain the scope of this PR - I am not making any actual change to the page. We can consider doing some of these optimizations after this PR has been landed in a separate PR